taf/api/roles.py

```diff
@@ -585,6 +587,10 @@ def _initialize_roles_and_keystore(
             taf_logger.info(
                 f"Keystore not specified. Using default location {default_keystore_path()}"
             )
+            if not Path(default_keystore_path()).is_dir():
```

Comment:
So this does raise an error like it's stated in the issue, but now that I'm testing this, I'm remembering that the idea was to generate the keystore files if the keystore does not exist. I think that in `create_repository`, if the keystore does not exist, we should ask the user if they want to generate new keystore or not.
For additional context - our production repositories currently share online keys (this is not necessarily the best practice, it's just way more convenient). This means that when David is creating a new repository, if the keystore is not found, he forgot to specify the keystore path. On the other hand, I envisioned this feature to be flexible enough so that you can create new keys, write them to disk, print them to the command line for the users to do whatever they want with them later, set up yubikeys etc.

Reply:
The code should already ask to generate keystore files. 
If no keystore directory or files are there, the command first generates keys-description.json through user inputs on the CLI, generates a keystore directory and puts it there, then check if other keystore files are there and if not will generate them. 

<img width="774" alt="Screenshot 2024-05-28 at 4 54 44 PM" src="https://github.com/openlawlibrary/taf/assets/71115970/e9ee7438-0186-4ad4-bcab-db53cd366d4e">

Reply:
That indeed is what should happen. However, if you do not specify keystore location, we get this error: `An error occurred while creating a new repository: Default Keystore Path D:\oll\library\oll-test-repos\test\keystore doesn't exist`

The default location is a directory named `keystore` inside the current working directory. Keystore path can be specified in that `keys-description` file, so we shouldn't turn keystore into a required argument. If it's not specified anywhere, we should print an error saying something like "Please provide keystore path". Looking at this, I would actually completely remove the concept of default keystore location. A while back, we thought that we could look for these keys inside the home directory, but the users found that to be very confusing

Reply:
Just added a new commit. 

With the new version keystore is not a required argument. If the keys-description file is provided, it gets the keystore directory from there. If the keys-description file isn't provided, a user can generate one via the CLI. 

When generating via CLI, it will prompt the user to enter the keystore directory. If they don't provide one, they get the error "Please provide keystore path." If they provided it but it doesn't exist, they get the error "Keystore path {keystore} does not exist."

If the keystore path is provided and exists but doesn't have the keystore files, the user can generate them via CLI.

2 questions:
1. If the keystore path is provided but doesn't exist, the user gets an error and the CLI command quits. Would it be better to create the directory if it doesn't exist? 
2. When a user generates a keys-description file via CLI, it doesn't get saved anywhere as a JSON file. Is this a problem?

Reply:
Yes, I think that we should create the directory if it does not already exist. Keen in mind that if keys-description is provided, it might not contain a keystore property.
I think that the idea was to print the generated keys-description to the command line. A user can copy-paste that to a file if they want to.

Reply:
Made fixes so it creates a directory. If only keys-description is provided and it doesn't point to a keystone, it outputs an error saying "Please provide keystore path."